### PR TITLE
[Home Page] Add Tokens Table

### LINKED ISSUE #39: [HomePage] - Tokens table UI
**Description**

1. Implement a mocked tokens table using existing components. For the pagination reuse the current component.
2. Implement an in-memory search filter that uses token name, symbol or hash as input. A library like [flexsearch](https://github.com/nextapps-de/flexsearch) could be used
3. Add the `Last 7 days` chart

> Original issue https://github.com/gnosis/gp-ui/issues/1070 by @ramirotw

### PR DESCRIPTION
**Pull request by matextrem**
# Summary

Closes #39 

Added Table tokens UI in the home page with mocked data.

image

This is a WIP version.

### CODE REVIEW

**Comment by matextrem** [comment]
@alongoni Feel free to add any change regarding styling 👍

**Review by alfetopito — COMMENTED** [review]
Couple of general comments about the UI

The `Top tokens` text looks like it could use a bit of padding/margin left, it's too close to the border IMO.
Also in the same image, the looking glass icon position is off. Checked both on chrome and FF
Screen Shot 2022-05-06 at 08 50 25

But actually, not sure we are releasing with a search bar in the first version, so we might want to hide that instead.

Third point, maybe it's too aggressive the switch to mobile view?
With `1180` pixels width it's already in a single column view and it does look weird with all that empty space:
Screen Shot 2022-05-06 at 08 58 08

Will review the code next

File: src/apps/explorer/components/TokensTableWidget/TokensTableWithData.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }, timeOutMs)
+
+    return (): void => {
+      clearTimeout(timeOutId)
+    }
+  }, [tokens, tokens?.length])
+
+  return isFirstRender || isFirstLoading ? (
+    <EmptyItemWrapper>
+      <Spinner spin size="3x" />
```
Comment (alfetopito) [inline src/apps/explorer/components/TokensTableWidget/TokensTableWithData.tsx:36]: Can you use Agustin's loader here instead?

File: src/apps/explorer/components/TokensTableWidget/PaginationTokensTable.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
+import React, { useContext } from 'react'
+import styled, { css } from 'styled-components'
+import { faChevronRight, faChevronLeft } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { media } from 'theme/styles/media'
+
+import { Dropdown, DropdownOption } from 'apps/explorer/components/common/Dropdown'
+import { TokensTableContext } from './context/TokensTableContext'
+
```
Comment (alfetopito) [inline src/apps/explorer/components/TokensTableWidget/PaginationTokensTable.tsx:9]: Could the pagination display and logic be somehow re-used between here and the orders page?
I have the impression that this is duplicated code.

**Comment by elena-zh** [comment]
Hey @matextrem , nice!
Some issues: 
1. I think, it would be nice to rewrite the message to 'No results found' or so 
no
2. I believe, it is not intentional, but Wrapped ETH symbol is WETH
symbol
3. I see an infinite loader when I change networks. Only refreshing the page stops it
loader when change
4. Table header should look a bit better in a mobile view
adjust header
5. It would be nice to write 'Tokens' from the lowercase letter, as we do for all the rest tables in the app
lower case
tanble names
6. The page is not auto-adjustable then change a screen width (might be related to #75 and #44 issues)
autoadjustable

7. 24H volume column name is shifted towards its values in the table
shifted

Thanks!

**Comment by ramirotw** [comment]
@matextrem @alongoni Found two small things to fix:

1. When switching between pages there's a small jump on the scroll, this is due to the logo's being loaded
2. When switching between networks, the table isn't cleared

https://user-images.githubusercontent.com/3328006/167397128-a668f7ac-256e-4816-b4de-883618e30041.mp4

**Comment by elena-zh** [comment]
Hey @matextrem , @alongoni , nice changes!
However, still some issues from my side: 
1. It would be nice to ad a period in the end of the sentence
add period
2. When hover on 1 line, all the rest lines are highlighted. See the video: https://watch.screencastify.com/v/JmAepcF1DH4vkcCpxDQ9
3. Can we make the scrollbar a bit nicer (add styles to it or so)?
nicer
4. Placeholder in the search field is missing in the tablet/mobile views
placeholder
5. Tokens table is blinking when change networks/reload the page/etc. Can we add something like a 'skeleton' effect to the table while it is being loaded (I reported #21  the issue for all the rest tables). So it will look more or less the same as loading effect on cards. 

Thanks!

**Comment by elena-zh** [comment]
I have added the p.5 into the #21 task  to fix it in another PR, and apply to all existing tables on Explorer. 
I still vote to implementing it before the Home page goes live to Prod.

**Comment by elena-zh** [comment]
Hey @alongoni , great job!

However, still some new issues and questions from my side: 
1. would it be nicer to add a separate column for tokens numbering, so the column name 'Name' will be displayed right above the token names.  ' WDYT?
separate
image

2. Can we center-align 'No results found' message as we do for all the rest tables in the app? 
image

3. I have noticed, that changes in this PR affected all existing tables, so tables names became center-aligned in mobile views
[transactions.txt](https://github.com/cowprotocol/explorer/files/8668799/transactions.txt)
left

Can we fix it? In addition, I think it would be nice to leave 'Top tokens' table name left-aligned as well to comply all the rest tables in the app. 

4. Can we move the placeholder text to another line as we do for the search field on small screen resolutions? 
image

Thanks

**Comment by alongoni** [comment]
> Hey @alongoni , great!
> 
> My last question is related to the search: the table does not show me a token when I'm on the 2nd page, and a token is listed on the 1st one. Will it be fixed in another PR (integration), or should be fixed here? (see the video with the issue: https://watch.screencastify.com/v/QCPvNa95pQRzdCj65Rsr)
> 
> Then, I noticed that the search does not support a partial match or combination of symbols that exist in a token name (ETH contains 'th', but search by 'th' dies not return any values). Is this an expected behavior? image
> 
> Thanks!

Thanks Elena! Maybe @matextrem can take a look.

**Comment by matextrem** [comment]
@elena-zh The first issue was addressed. Regarding the second one, it is intended since we are comparing string as `startsWith`. I think we could change it to do something like `contains` if it is needed.

**Review by elena-zh — APPROVED** [review]
Changes LGTM now!

I have created a separate task to enhance the search feature #91 (in case we decide to fix it later)

**Comment by elena-zh** [comment]
Just to recall: I think, we still need to fix an issue with loading table effects #21 to correspond all the rest elements on the page 
Also, I always face issue with auto-resizing page elements when I change a screen width  #75

**Review by ramirotw — APPROVED** [review]

**Review by alongoni — APPROVED** [review]
🚀 Looks great!

**Pull request merged**